pycode/memilio-epidata/memilio/epidata/getCaseData.py

```diff
@@ -0,0 +1,676 @@
+    gbAllC = df.groupby([dateToUse, IdLandkreis]).\
+        agg({AnzahlFall: sum, AnzahlTodesfall: sum, AnzahlGenesen: sum})
+    gbAllC_cs = gbAllC.groupby(level=1).cumsum().reset_index()
+
+    # output
+    if split_berlin:
+        filename = 'all_county_split_berlin'
+    else:
+        filename = 'all_county'
+    if rep_date:
+        filename_orig = filename + '_repdate'
+    else:
+        filename_orig = filename
+    gd.write_dataframe(gbAllC_cs, directory,
+                       prefix + filename_orig, file_format)
+    if impute_dates or moving_average > 0:
+        gbAllC_cs = modifyDataframeSeries.impute_and_reduce_df(
+            gbAllC_cs,
+            {dd.EngEng["idCounty"]: sorted(set(df[dd.EngEng["idCounty"]].unique()))},
+            ['Confirmed', 'Deaths', 'Recovered'],
+            impute='forward', moving_average=moving_average)
+        filename = gd.append_filename(filename, impute_dates, moving_average)
+        if rep_date:
+            filename = filename + '_repdate'
+        gd.write_dataframe(gbAllC_cs, directory,
+                           prefix + filename, file_format)
+
+    ######### Data whole Germany different gender ##################
+
+    # infected (incl recovered), deaths and recovered together
+
+    gbAllG = df.groupby([Geschlecht, dateToUse]) \
+        .agg({AnzahlFall: sum, AnzahlTodesfall: sum, AnzahlGenesen: sum})
+
+    gbAllG_cs = gbAllG.groupby(level=0).cumsum().reset_index()
+
+    # output
+    filename = 'all_gender'
+    if rep_date:
+        filename_orig = filename + '_repdate'
+    else:
+        filename_orig = filename
+    gd.write_dataframe(gbAllG_cs, directory,
+                       prefix + filename_orig, file_format)
+    if impute_dates or moving_average > 0:
+        gbAllG_cs = modifyDataframeSeries.impute_and_reduce_df(
+            gbAllG_cs,
+            {dd.EngEng["gender"]: list(df[dd.EngEng["gender"]].unique())},
+            ['Confirmed', 'Deaths', 'Recovered'],
+            impute='forward', moving_average=moving_average)
+        filename = gd.append_filename(filename, impute_dates, moving_average)
+        if rep_date:
+            filename = filename + '_repdate'
+        gd.write_dataframe(gbAllG_cs, directory,
+                           prefix + filename, file_format)
+
+    if make_plot:
+        df.groupby(Geschlecht) \
+            .agg({AnzahlFall: sum, AnzahlTodesfall: sum, AnzahlGenesen: sum}) \
+            .plot(title='COVID-19 infections, deaths, recovered', grid=True,
+                  kind='bar')
+        plt.tight_layout()
+        plt.show()
+
+    ############################# Gender and State ######################################################
+
+    # infected (incl recovered), deaths and recovered together
+
+    gbAllGState = df.groupby([dateToUse, IdBundesland, Geschlecht]) \
+        .agg({AnzahlFall: sum, AnzahlTodesfall: sum, AnzahlGenesen: sum})
+    gbAllGState_cs = gbAllGState.groupby(level=[1, 2]).cumsum().reset_index()
+
```

Comment:
Thank you for the link. This helped me to understand the what kind of levels are meant by this notation. First changing this raised an error in the tests. I fixed it with try and error by changing the order in the group by statement. Afterwards I checked with a script whether the cumulative sum is done along the right axes by checking that the cumulative sum always increases. I wanted to write a test for this kind of error but this turned out to be more difficult than I thougt since in the test set which already exists there are only one or two rows per county. I think, this is too few data to detect when the cumulative sum is computed over the wrong columns by checking whether the cumulative sum is increasing. Maybe a new issue should be opened for implementing a test that detects such an error.
I will change the notation according to the example in the vaccination data file which was better understadable for me and seems to be not so likely to introduce errors of this kind.